### PR TITLE
chore(ci_visibility): improve logging of CIVisibility and pytest ddtrace plugin

### PR DESCRIPTION
**Pull request by romainkomorndatadog**
This introduces better logging (on the back of the new CI Visibility logger) to provide more feedback to users.

Log example:
```
pytest --ddtrace
[Datadog CI Visibility] INFO     ddtrace.internal.ci_visibility.recorder:recorder.py:183 Service: flask (env: itr-on)
[Datadog CI Visibility] INFO     ddtrace.internal.ci_visibility.recorder:recorder.py:184 Requests mode: agentless
[Datadog CI Visibility] INFO     ddtrace.internal.ci_visibility.recorder:recorder.py:185 Git metadata upload enabled: True
[Datadog CI Visibility] INFO     ddtrace.internal.ci_visibility.recorder:recorder.py:186 API-provided settings: coverage collection: True
[Datadog CI Visibility] INFO     ddtrace.internal.ci_visibility.recorder:recorder.py:187 API-provided settings: Intelligent Test Runner: True, test skipping: True
[Datadog CI Visibility] INFO     ddtrace.internal.ci_visibility.recorder:recorder.py:192 Detected configurations: {'os.architecture': 'arm64', 'os.platform': 'Darwin', 'os.version': '23.2.0', 'runtime.name': 'CPython', 'runtime.version': '3.10.12'}
[Datadog CI Visibility] WARNING  ddtrace.internal.ci_visibility.recorder:recorder.py:199 CODEOWNERS file is not available
[Datadog CI Visibility] INFO     ddtrace.internal.ci_visibility.recorder:recorder.py:546 Intelligent Test Runner skipping level: test
[Datadog CI Visibility] INFO     ddtrace.internal.ci_visibility.recorder:recorder.py:547 Skippable items fetched: 295
[Datadog CI Visibility] INFO     ddtrace.internal.ci_visibility.recorder:recorder.py:505 Final settings: coverage collection: True, test skipping: True
```

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

File: ddtrace/contrib/pytest/plugin.py [hunk 1]
```diff
@@ -563,6 +563,8 @@ def pytest_collection_modifyitems(session, config, items):
             for item_to_skip in items_to_skip:
                 item_to_skip.add_marker(skip)
 
+        log.info("Intelligent Test Runner ")
+
```
Comment (romainkomorndatadog) [inline ddtrace/contrib/pytest/plugin.py:567]:
```suggestion
```

**Comment by romainkomorndatadog** [comment]
Going back to draft while I fix tests.

**Review by nikita-tkachenko-datadog — APPROVED** [review]

**Comment by romainkomorndatadog** [comment]
Another day, another attempt at resolving merge conflicts in the GitHub UI... 🤞

**Review by majorgreys — COMMENTED** [review]
My concern here is mostly with the user impact of these changes. Being changes to the product logs, is the idea that this is purely internal? Do users need to be informed about any of this change?

**Comment by romainkomorndatadog** [comment]
> My concern here is mostly with the user impact of these changes. Being changes to the product logs, is the idea that this is purely internal? Do users need to be informed about any of this change?

TL;DR: I thought about it while working on this PR and decided not to inform users of the extra logging (but the original "there's a new logger **is** announced in [the 2.5.0 release notes](https://github.com/DataDog/dd-trace-py/releases/tag/v2.5.0). I'm open to changing it though.

It's user facing (so not purely internal). I self-debated this, because `pytest` already logs some things by default, but at a higher log level, and it's definitely less verbose if `--cli-log-level` isn't passed in at all.

I could make the default log level for this higher (eg: warning, as it ends up being by default), which would only introduce the formatting change where
```
CODEOWNERS file is not available
```
becomes
```
[Datadog CI Visibility] WARNING  ddtrace.internal.ci_visibility.recorder:recorder.py:199 CODEOWNERS file is not available
```
but IMO, this is a "`O(1)`" change. We don't `log.info()` anything on every test, so we're not adding "unbound" growth to log sizes.

As I mentioned in the TL;DR, the new logger itself is mentioned in https://github.com/DataDog/dd-trace-py/releases/tag/v2.5.0 , so "we're logging more with this new logger" didn't feel necessary, but I'm open to adding a release note.

**Review by majorgreys — APPROVED** [review]
